### PR TITLE
Fix(continuum): update RGs without O(N^2) complexity.

### PR DESCRIPTION
**Pull request by 3Hren**
Boost accumulate is **shit**.

It works through passing the accumulator value by value each iteration. Each iteration we have linear increasing number of copying continuums, which itself contains hundreds of strings.

This commit forces the continuum to be move only and applies emplace semantics for std::map with routing groups.

For example we have 4 RG with 256, 256, 256 and 512 points respectively. That's the result of 3 x {"app": 1} and 1 x {"app": 2, "other": 1} groups.
For this case after construction and one full update we have ~11776 continuum `element_t` copies which involves string reallocations.
With 8 RGs on a single refresh we suffer from ~77312 allocations and so on.

In our production there are at least 60 RGs ...

Unfortunately we can't completely get rid of copying because of atomic RG refresh, but it's O(N).

After this commit we have linear number number of copies on refresh.

### CODE REVIEW

File: src/service/locator.cpp [hunk 1]
```diff
@@ -532,9 +542,9 @@ locator_t::on_refresh(const std::vector<std::string>& groups) {
         } else {
             COCAINE_LOG_INFO(m_log, "removing routing group");
         }
+    }
 
-        return accumulator;
-    });
+    m_rgs.unsafe() = std::move(accumulator);
```
Comment (3Hren) [inline src/service/locator.cpp]: I don't think it's okay, but this is how it worked previously.

**Comment by 3Hren** [comment]
@kobolog @noxiouz @antmat PTAL

**Comment by 3Hren** [comment]
The same problem (but with less scale) lies several lines below. I'll fix them too today.

**Comment by noxiouz** [comment]
LGTM, except the last comment =)

**Comment by 3Hren** [comment]
Squashed

**Comment by noxiouz** [comment]
LGTM is still here.

**Comment by kobolog** [comment]
VS https://github.com/cocaine/cocaine-core/pull/246

**Pull request closed without merging**